bundles/org.openhab.binding.weathercompany/src/main/java/org/openhab/binding/weathercompany/internal/handler/WeatherCompanyHandler.java

```diff
@@ -0,0 +1,744 @@
+        request.header(HttpHeader.ACCEPT_ENCODING, "gzip");
+
+        String errorMsg;
+        try {
+            ContentResponse contentResponse = request.send();
+            switch (contentResponse.getStatus()) {
+                case HttpStatus.OK_200:
+                    String response = contentResponse.getContentAsString();
+                    String cacheControl = contentResponse.getHeaders().get(HttpHeader.CACHE_CONTROL);
+                    logger.debug("Cache-Control header is {}", cacheControl);
+                    return response;
+                case HttpStatus.NO_CONTENT_204:
+                    errorMsg = "HTTP response 400: No content. Check configuration";
+                    break;
+                case HttpStatus.BAD_REQUEST_400:
+                    errorMsg = "HTTP response 400: Bad request";
+                    break;
+                case HttpStatus.UNAUTHORIZED_401:
+                    errorMsg = "HTTP response 401: Unauthorized";
+                    break;
+                case HttpStatus.FORBIDDEN_403:
+                    errorMsg = "HTTP response 403: Invalid API key";
+                    break;
+                case HttpStatus.NOT_FOUND_404:
+                    errorMsg = "HTTP response 404: Endpoint not found";
+                    break;
+                case HttpStatus.METHOD_NOT_ALLOWED_405:
+                    errorMsg = "HTTP response 405: Method not allowed";
+                    break;
+                case HttpStatus.NOT_ACCEPTABLE_406:
+                    errorMsg = "HTTP response 406: Not acceptable";
+                    break;
+                case HttpStatus.REQUEST_TIMEOUT_408:
+                    errorMsg = "HTTP response 408: Request timeout";
+                    break;
+                case HttpStatus.INTERNAL_SERVER_ERROR_500:
+                    errorMsg = "HTTP response 500: Internal server error";
+                    break;
+                case HttpStatus.BAD_GATEWAY_502:
+                case HttpStatus.SERVICE_UNAVAILABLE_503:
+                case HttpStatus.GATEWAY_TIMEOUT_504:
+                    errorMsg = String.format("HTTP response %d: Service unavailable or gateway issue",
+                            contentResponse.getStatus());
+                    break;
+                default:
+                    errorMsg = String.format("HTTP GET failed: %d, %s", contentResponse.getStatus(),
+                            contentResponse.getReason());
+                    break;
+            }
+        } catch (TimeoutException e) {
+            errorMsg = "TimeoutException: Call to Weather Company API timed out";
+        } catch (ExecutionException e) {
+            errorMsg = String.format("ExecutionException: %s", e.getMessage());
+        } catch (InterruptedException e) {
+            errorMsg = String.format("InterruptedException: %s", e.getMessage());
+        }
+        markThingOffline(errorMsg);
+        logger.debug(errorMsg);
```

Comment:
AFAIK this is warning for the SAT checks

Reply:
SAT isn't complaining, so I'll leave it if you're ok. Let me know if you'd prefer I change/remove it.

Reply:
After I rebased to the latest master, it started throwing errors. It's fixed now.